src/Main.elm

```diff
@@ -0,0 +1,197 @@
+module Main exposing (..)
+
+import Html exposing (Html, div, h2, button, text, a)
+import Html.Attributes exposing (href, id)
+import Html.Events exposing (onClick)
+import Movie exposing (Movie, movieCard)
+import MovieList exposing (..)
+import AppCss.Helpers exposing (class)
+import AppCss exposing (..)
+import Time.Date as Date exposing (date)
+import Random
+import Random.List as RandList
+import Markdown
+
+
+main : Program Never Model Msg
+main =
+    Html.program
+        { init = init
+        , view = view
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+
+-- Model
+
+
+type alias Model =
+    { unwatched : List Movie
+    , watched : List Movie
+    , selectedMovie : MovieSelection
+    }
+
+
+type MovieSelection
+    = NotSelected
+    | Selected Movie
+    | NothingToSelect
+
+
+init : ( Model, Cmd Msg )
+init =
+    let
+        watchDate movie =
+            Movie.watchDate movie
+                |> Maybe.withDefault (date 1800 1 1)
+    in
+        { unwatched =
+            MovieList.movies
+                |> List.filter (not << Movie.isWatched)
+                |> List.sortBy .year
+        , watched =
+            MovieList.movies
+                |> List.filter Movie.isWatched
+                |> List.sortWith
+                    (\m1 m2 ->
+                        Date.compare (watchDate m1) (watchDate m2)
+                    )
+        , selectedMovie = NotSelected
+        }
+            ! []
+
+
+
+-- Update
+
+
+type Msg
+    = SelectMovie
+    | MovieSelected ( Maybe Movie, List Movie )
+
+
+update : Msg -> Model -> ( Model, Cmd Msg )
+update msg model =
+    case msg of
+        SelectMovie ->
+            ( model
+            , Random.generate
+                MovieSelected
+                (RandList.choose model.unwatched)
+            )
```

Comment:
This might be more information than you need, but I want to make sure I don't miss anything.

Yes, `SelectMovie` is an incoming `Msg` that indicates the user has requested that a new random movie be selected.  The contract for `update` is that it returns `( Model, Cmd Msg)`, which is a data structure called a tuple. The contents of the tuple are the new state of the application (`Model`), and a description of any side-effects that need to be triggered (`Cmd Msg`).  In this case, the model is unchanged, so our tuple starts be returning the input model: `(model, ...` and triggers a side effect of choosing a random movie from the `model.unwatched`list. This:

`Random.generate MovieSelected (RandList.choose model.unwatched)` 

says "Generate a random value by choosing an item from `model.unwatched` and return the result in a `MovieSelected` message. and fills the `..., Cmd Msg)` part of the `update` contract.

I suspect the parens of the tuple might be the point of confusion here. Tuples are a little confusing at first, because it means parens are used in code both for order of operations and a data structure, but in general if you see `(foo, bar)` or `(foo, bar, baz, ...)` you're looking at a tuple.  The clarity of this code is probably not helped by the fact that elswhere in update I used the `!` helper to generate a tuple with no commands.  I could have written the code here as:

`model ! [Random.generate MovieSelected (RandList.choose model.unwatched)]`

I've been struggling a bit with deciding how often to use the `!` helper vs just explicitly creating the tuple, and which approach is clearer/easier. It was probably a bad idea to mix and match here.